mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSNode.kt

```diff
@@ -220,8 +220,8 @@ data class MPSNode(val node: SNode) : IDefaultNodeAdapter, IReplaceableNode {
                 ?: SReferenceLinkAdapterById(SReferenceLinkId.deserialize(link.getUID()), "")
         }
 
-        val targetNode = target?.let { getArea().resolveNode(it.reference) } as MPSNode
-        node.setReferenceTarget(refLink, targetNode.node)
+        val targetNode = target?.let { getArea().resolveNode(it.reference) as MPSNode? }
```

Comment:
This line didn't make sense before and still doesn't. Why do we try to resolve a node that is already an `INode`? Either the caller passes a valid `MPSNode` or this method has to fail. This is the wrong place for re-resolving nodes.

Reply:
Done